File: app/src/main/java/fr/xgouchet/chronorg/data/ioproviders/IOProvider.java
package fr.xgouchet.chronorg.data.ioproviders;

import android.database.Cursor;
import android.support.annotation.NonNull;

import fr.xgouchet.chronorg.data.queriers.ContentQuerier;
import fr.xgouchet.chronorg.data.readers.BaseCursorReader;
import fr.xgouchet.chronorg.data.writers.BaseContentValuesWriter;

/**
 * @author Xavier Gouchet
 */
public interface IOProvider<T> {

    @NonNull BaseCursorReader<T> provideReader(@NonNull Cursor cursor);

    @NonNull BaseContentValuesWriter<T> provideWriter();

    @NonNull ContentQuerier<T> provideQuerier();
}

File: khronorg/src/main/java/fr/xgouchet/khronorg/data/ioproviders/IOProvider.kt
package fr.xgouchet.khronorg.data.ioproviders

import android.database.Cursor
import fr.xgouchet.khronorg.data.queriers.ContentQuerier
import fr.xgouchet.khronorg.data.readers.Reader
import fr.xgouchet.khronorg.data.writers.Writer

/**
 * @author Xavier F. Gouchet
 */
interface IOProvider<T> {

    fun provideReader(cursor: Cursor): Reader<T>

    val writer: Writer<T>

    val querier: ContentQuerier<T>
}
